design/metadata-handling.md

```diff
@@ -0,0 +1,668 @@
+- vhostuser
+- vif
+- phy
+
+The **links/ethernet/macAddress** object can be one of:
+
+- **string**: with the desired Mac given as a string
+- **bareMetalHostMACByName**: with the interface name from BareMetalHost
+  hardware details.
+
+The **links/bond** object contains the following:
+
+- **id**: Interface name
+- **mtu**: Interface MTU
+- **macAddress**: an object to render the MAC Address
+- **bondMode**: The bond mode
+- **bondLinks** : a list of links to use for the bond
+
+The **links/bond/bondMode** can be one of :
+
+- 802.1ad
+- balance-rr
+- active-backup
+- balance-xor
+- broadcast
+- balance-tlb
+- balance-alb
+
+The **links/vlan** object contains the following:
+
+- **id**: Interface name
+- **mtu**: Interface MTU
+- **macAddress**: an object to render the MAC Address
+- **vlanId**: The vlan ID
+- **vlanLink** : The link on which to create the vlan
+
+##### The networks specifications
+
+The object for the **networks** section list can be one of:
+
+- **ipv4**: an ipv4 static allocation
+- **ipv4Dhcp**: an ipv4 dhcp based allocation
+- **ipv6**: an ipv6 static allocation
+- **ipv6Dhcp**: an ipv6 dhcp based allocation
+- **ipv6Slaac**: an ipv6 slaac based allocation
+
+The **networks/ipv4** object contains the following:
+
+- **id**: the network name
+- **link**: The name of the link to configure this network for
+- **ipAddress**: the IP address object
+- **netmask**: the netmask, in an integer format
+- **routes**: the list of route objects
+
+The **networks/ipv4/ipAddress** is an address object containing:
+
+- **start**: the start IP address
+- **end**: The end IP address
+- **subnet**: The subnet in a CIDR notation "X.X.X.X/X"
+- **step**: the step between IP addresses
+
+If the **subnet** is specified, then **start** and **end** are not required and
+reverse, if **start** and **end** are specified, then **subnet** is not required
+
+The **networks/ipv4/routes** is a route object containing:
+
+- **network**: the subnet to reach
+- **netmask**: the mask of the subnet as integer
+- **gateway**: the gateway to use
+- **services**: a list of services object as defined later
+
+The **networks/ipv4Dhcp** object contains the following:
+
+- **id**: the network name
+- **link**: The name of the link to configure this network for
+- **routes**: the list of route objects
+
+The **networks/ipv6** object contains the following:
+
+- **id**: the network name
+- **link**: The name of the link to configure this network for
+- **ipAddress**: the IP address object
+- **netmask**: the netmask, in an integer format
+- **routes**: the list of route objects
+
+The **networks/ipv6Dhcp** object contains the following:
+
+- **id**: the network name
+- **link**: The name of the link to configure this network for
+- **routes**: the list of route objects
+
+The **networks/ipv6Slaac** object contains the following:
+
+- **id**: the network name
+- **link**: The name of the link to configure this network for
+- **routes**: the list of route objects
```

Comment:
The first time I read through this section I thought we didn't need 5 different types (ipv4, ipv6, ipv4Dhcp, ipv6Dhcp, ipv6Slaac) because they're all just variations and we could have a "type" field. But after considering it, I like using the separate types because it makes API validation a bit easier and it's easier to describe which fields are needed for each type.

Reply:
I followed the exact same train of thoughts when writing it :)